base_product_mass_addition/models/product_product.py

```diff
@@ -0,0 +1,114 @@
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    qty_to_process = fields.Float(
+        compute="_compute_process_qty",
+        inverse="_inverse_set_process_qty",
+        help="Set this quantity to create a new line "
+        "for this product or update the existing one.",
+    )
+    quick_uom_category_id = fields.Many2one(
+        "uom.category", compute="_compute_quick_uom_info"
+    )
+    quick_uom_id = fields.Many2one(
+        "uom.uom",
+        domain="[('category_id', '=', quick_uom_category_id)]",
+        compute="_compute_quick_uom_info",
+        inverse="_inverse_set_process_qty",
+    )
+
+    def _inverse_set_process_qty(self):
+        parent = self.pma_parent
+        if parent:
+            for product in self:
+                quick_line = parent._get_quick_line(product)
+                if quick_line:
+                    parent._update_quick_line(product, quick_line)
+                else:
+                    parent._add_quick_line(product, quick_line._name)
+
+    def _DISABLED_modified(self, fnames, create=False, before=False):
+        # OVERRIDE to supress LOG_ACCESS_COLUMNS writes if we're only writing on quick
+        # magic fields, as they could lead to concurrency issues.
+        #
+        # Moreover, from a functional perspective, these magic fields aren't really
+        # modifying the product's data so it doesn't make sense to update its metadata.
+        #
+        # We achieve it by reverting the changes made by ``write`` [^1], before [^2]
+        # reaching any explicit flush [^3] or inverse computation [^4].
+        #
+        # [^1]:
+        # https://github.com/odoo/odoo/blob/3991737a53e75398fcf70b1924525783b54d256b/odoo/models.py#L3778-L3787 # noqa: B950
+        # [^2]:
+        # https://github.com/odoo/odoo/blob/3991737a53e75398fcf70b1924525783b54d256b/odoo/models.py#L3882 # noqa: B950
+        # [^3]:
+        # https://github.com/odoo/odoo/blob/3991737a53e75398fcf70b1924525783b54d256b/odoo/models.py#L3885 # noqa: B950
+        # [^4]:
+        # https://github.com/odoo/odoo/blob/f74434c6f4303650e886d99fb950c763f2d4cc6e/odoo/models.py#L3703 # noqa: B950
+        #
+        # Basically, if all we're modifying are quick magic fields, and we don't have
+        # any other column to flush besides the LOG_ACCESS_COLUMNS, clear it.
+        quick_fnames = ("qty_to_process", "quick_uom_id")
+        if (
+            self
+            and fnames
+            and any(quick_fname in fnames for quick_fname in quick_fnames)
+        ):
+            for record in self.filtered("id"):
+                towrite = self.env.all.towrite[self._name]
+                vals = towrite[record.id]
+                if not vals:  # pragma: no cover
+                    continue
+                if all(fname in LOG_ACCESS_COLUMNS for fname in vals.keys()):
+                    towrite.pop(record.id)
+        return super().modified(fnames, create=create, before=before)
```

Comment:
Hi @ivantodorovich. Thanks a lot for your time. (analysis and code proposal)
I tested your implementation (see last fixup commit).

- positive point : it works ! I mean, with purchase_quick installed, if change the field qty_to_process field, it is writing on pos.order.line, but doesn't change write_date field :+1: 
- negative point : the test fails. see https://github.com/OCA/product-attribute/actions/runs/7972609836/job/21764713992?pr=1512#step:8:518 :-1: 

Any idea ? 

```
   File "/__w/product-attribute/product-attribute/base_product_mass_addition/tests/test_product_mass_addition.py", line 55, in test_quick_should_not_write_on_product
    self.assertEqual(self.product.write_uid, user_demo)
AssertionError: res.users(1,) != res.users(6,)
```

Reply:
Hmm, I would put a breakpoint there: 

```
            if all(
                field.name in LOG_ACCESS_COLUMNS
                for field in self.env.cache.get_dirty_fields()
            )
```

And then check why the condition is not matched.

It could be that there's an extra dirty fields in there, during tests. In which case we might need to add some explicit flushes in tests.

---

Alternatively, it may be due to the simplification of the new implementation:

> It's not as smart as the previous one, but it might work in our specific use case (we won't be updating other product fields anyway)

Maybe there are other dirty products in the cache. In which case the solution might be also to flush during tests, or improve the code to restore the previous record-by-record dirty check -- but to do that we'd have to mess with the cache internals as there's no public api for this AFAICS

Reply:
> And then check why the condition is not matched.

i put a breakpoint, and the call of clear_dirty_field is done. 
So, self.env.cache.get_dirty_fields() is [write_uid, write_date] at the beginning of the call of ``modified`` and when super is executed, it is []. That looks correct. _However_ via test, it doesn't work.

Reply:
Hi as said in the original PR the tests does not work because it is not possible to change the value of all log_access field.

I wrote a PR to your branch @legalsylvain to fix the test https://github.com/grap/product-attribute/pull/2